server/src/main/java/org/elasticsearch/action/bulk/BulkShardRequest.java

```diff
@@ -31,10 +31,12 @@
 import java.io.IOException;
 import java.util.HashSet;
 import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 public class BulkShardRequest extends ReplicatedWriteRequest<BulkShardRequest> {
 
-    private BulkItemRequest[] items;
+    private final AtomicBoolean bytesAccounted = new AtomicBoolean(false);
```

Comment:
I'll look at this tomorrow. The primary reason I did not do it that way is if a request was sent to another node and then relocated back to the coordinating node it would not be accounted. But that does not really matter with the current design.